File: eMovieFinder/eMovieFinder.Model/SearchObjects/OrderSearchObject.cs
namespace eMovieFinder.Model.SearchObjects
{
    public class OrderSearchObject : BaseSearchObject
    {
        public int? UserId { get; set; }
        public bool? IsUserIncluded { get; set; }
        public bool? IsOrderMovieIncluded { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_mobile/lib/src/models/searchobjects/order_search_object.dart
import 'package:json_annotation/json_annotation.dart';

import 'base_search_object.dart';

part 'order_search_object.g.dart';

@JsonSerializable()
class OrderSearchObject extends BaseSearchObject {
  final int? userId;
  final bool? isUserIncluded;
  final bool? isOrderMovieIncluded;

  OrderSearchObject({
    super.pageNumber,
    super.pageSize,
    super.orderBy,
    super.isDescending,
    this.userId,
    this.isUserIncluded,
    this.isOrderMovieIncluded
  });

  factory OrderSearchObject.fromJson(Map<String, dynamic> json) => _$OrderSearchObjectFromJson(json);
  Map<String, dynamic> toJson() => _$OrderSearchObjectToJson(this);
}